### PR TITLE
Validations model specs

### PR DESCRIPTION
**Pull request by EzekielUtshudi**
### **Changes in this pull request :**

- [ ]  Added validations to User and Post models.
- [ ]  Added unit test for User, Post, Comment, and Like models.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,11 +40,12 @@ To get a local copy up and running follow these simple example steps.
 
 ## Authors
```
Comment (hiromataba) [inline README.md:72]: - [x] Please add information to run your project locally to set it up and much more that can make your README very professional

File: db/migrate/20220705134148_remove_ref_from_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class RemoveRefFromComments < ActiveRecord::Migration[7.0]
+  def change
```
Comment (hiromataba) [inline db/migrate/20220705134148_remove_ref_from_comments.rb]: - [ ] When running the migration I am getting the following error please fix that 👇 the next reviewer will make sure you tests are passing after you fix the issue 😉 

image

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @EzekielUtshudi     :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220705115226_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+
+      t.timestamps
+    end
+  end
+end
```
Comment (codepantha) [inline db/migrate/20220705115226_create_users.rb]: - [ ] An error occurs at the point when the Users table is being created while running your migrations. Please fix it 🙏. I have attached the error below 👇 
image
The next reviewer will proceed to check other aspects of your project after you have fixed this issue

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️  

## Hi @EzekielUtshudi 👋, 

Well done implementing the required changes but there is still a bit of work to be done before you can proceed to the next milestone. You're almost there 👏 



### Highlights ⭐ 
✔️ Descriptive PR
✔️ Good commit messages
✔️ Linters checks passing

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220705115226_create_users.rb (hunk 3 above)
Comment (EzekielUtshudi) [reply db/migrate/20220705115226_create_users.rb]: HEY @codepantha ,sorry but i think this errors are on your side

**Comment by EzekielUtshudi** [comment]
Screenshot_20220708_225848
 ABOVE is the screenshot of my tests working

File: db/migrate/20220705115226_create_users.rb (hunk 3 above)
Comment (EzekielUtshudi) [reply db/migrate/20220705115226_create_users.rb]: screenshot 1
Please see the attached images of my tests and migratio
screenshot 2
n

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @EzekielUtshudi 👋🏻, 

Great job on making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional ♻️ 

- [ ] Kindly add the `tmp` folder to the `.gitignore`. It helps to ignore files that may make the repository large. in file size.


Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**